src/plugins/intel_gpu/src/graph/include/multiclass_nms_inst.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+#include "intel_gpu/primitives/multiclass_nms.hpp"
+#include "primitive_inst.h"
+
+namespace cldnn {
+template <>
+struct typed_program_node<multiclass_nms> : public typed_program_node_base<multiclass_nms> {
+    using parent = typed_program_node_base<multiclass_nms>;
+
+public:
+    typed_program_node(std::shared_ptr<primitive> prim, program& prog)
+        : parent(prim, prog),
+          has_roisnum_(this->get_primitive()->has_roisnum) {}
+
+    program_node& input() const {
+        return boxes();
+    }
+
+    program_node& boxes() const {
+        return get_dependency(0);
+    }
+
+    program_node& scores() const {
+        return get_dependency(1);
+    }
+
+    bool has_roisnum() const {
+        return has_roisnum_;
+    }
+
+    program_node& roisnum() const {
+        if (!has_roisnum_)
+            throw std::runtime_error("there is no roisnum input");
+        return get_dependency(2);
+    }
+
+    program_node& output_selected_indices() const {
+        return get_dependency(input_count());
+    }
+    program_node& output_selected_num() const {
+        return get_dependency(input_count() + 1);
+    }
+
+private:
+    int input_count() const {
+        return 2 + (has_roisnum_ ? 1 : 0);
+    }
+
+    bool has_roisnum_;
```

Comment:
why we need to duplicate this value if we always can extract it from primitive and use has_roisnum() method where it's needed?

Reply:
Done.